packages/block-editor/src/components/line-height-control/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { TextControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import {
+	BASE_DEFAULT_VALUE,
+	RESET_VALUE,
+	STEP,
+	useIsLineHeightControlsDisabled,
+	useLineHeightControlState,
+	isLineHeightDefined,
+} from './utils';
+
+export default function LineHeightControl( {
+	className,
+	style = {},
+	...props
+} ) {
+	const [ lineHeight, setLineHeight ] = useLineHeightControlState();
+	const isDisabled = useIsLineHeightControlsDisabled();
+	const isDefined = isLineHeightDefined( lineHeight );
+
+	// Don't render the controls if disabled by editor settings
+	if ( isDisabled ) {
+		return null;
+	}
+
+	const handleOnChange = ( nextValue ) => {
+		// Set the next value without modification if lineHeight has been defined
+		if ( isDefined ) {
+			setLineHeight( nextValue );
+			return;
+		}
+
+		// Otherwise...
+		let adjustedNextValue = nextValue;
+
+		switch ( nextValue ) {
+			case `${ STEP }`:
+				// Increment by step value
+				adjustedNextValue = BASE_DEFAULT_VALUE + STEP;
+				break;
+			case '0':
```

Comment:
If I write a value of 0 directly we assume right away another value as if the user was decreasing the value.
If I write a value of 0.1 directly we assume right away another value as if the user was increasing the value.

Maybe in the event that is triggered, we can check if the controls up/down are being used instead of using the value as a way to check?

Reply:
Oh, good call. This one is tricky. We can capture `UP`/`DOWN` on keyboard, but not so much clicking `up`/`down` in the `input` control.

Investigating...

Reply:
@jorgefilipecosta Haii! I just pushed a change to account for this logic.

Typing in `0` (zero) has now been accounted for.

Some good news is, the user can't physically trigger the `0.1` case by typing. This flow can only be triggered by either pressing `UP` or  clicking the `up` arrow in the `input`.